src/EditorFeatures/CSharpTest/Completion/CompletionProviders/ExtensionMethodImportCompletionProviderTests.cs

```diff
@@ -0,0 +1,1071 @@
+                    new List<string>()
+                    { "string", "String", "System.String"},
+                    new List<string>()
+                    { "object", "Object", "System.Object"},
+                };
+
+                var arraySuffixes = new[] { "", "[]", "[,]" };
+
+                foreach (var group in predefinedTypes)
+                {
+                    foreach (var type1 in group)
+                    {
+                        foreach (var type2 in group)
+                        {
+                            foreach (var suffix in arraySuffixes)
+                            {
+                                yield return new List<object>() { type1 + suffix, type2 + suffix };
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        private static string GetMarkup(string current, string referenced, ReferenceType refType,
+                                        string currentLanguage = LanguageNames.CSharp,
+                                        string referencedLanguage = LanguageNames.CSharp)
+            => refType switch
+            {
+                ReferenceType.None => CreateMarkupForSingleProject(current, referenced, currentLanguage),
+                ReferenceType.Project => GetMarkupWithReference(current, referenced, currentLanguage, referencedLanguage, true),
+                ReferenceType.Metadata => GetMarkupWithReference(current, referenced, currentLanguage, referencedLanguage, false),
+                _ => null,
+            };
+
+        public static IEnumerable<object[]> BuiltInTypesWithReferenceTypeData
+            => CombineWithReferenceTypeData(BuiltInTypes);
+
+        [MemberData(nameof(BuiltInTypesWithReferenceTypeData))]
+        [Theory, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task TestPredefinedType(string type1, string type2, ReferenceType refType)
+        {
+            var file1 = $@"
+using System;
+
+namespace Foo
+{{
+    public static class ExtensionClass
+    {{
+        public static bool ExtentionMethod(this {type1} x)
+            => true;
+    }}
+}}";
+            var file2 = $@"
+using System;
+
+namespace Baz
+{{
+    public class Bat
+    {{
+        public void M({type2} x)
+        {{
+            x.$$
+        }}
+    }}
+}}";
+
+            var markup = GetMarkup(file2, file1, refType);
+
+            await VerifyTypeImportItemExistsAsync(
+                 markup,
+                 "ExtentionMethod",
+                 glyph: (int)Glyph.ExtensionMethodPublic,
+                 inlineDescription: "Foo");
+        }
+
+        [MemberData(nameof(ReferenceTypeData))]
+        [Theory, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task UsingAliasInDeclaration(ReferenceType refType)
+        {
+            var file1 = $@"
+using System;
+using MyInt = System.Int32;
+
+namespace Foo
+{{
```

Comment:
Why do we need double braces? It seems that existing tests use single ones.

Reply:
It's copied from previous test. Removed